form.js

```diff
@@ -22,137 +22,115 @@ define('form', ['doc'], function($) {
+	};
 
-			if (value && value.length < length) {
-				var parent = element.parentElement;
-				appendMessageWithArgs($(parent), validationMessages.min, length);
-				valid = false;
-			}
-		});
+	var getMaxOrMinMessage = function($field, condition) {
+		var fieldValue = $field.val(),
+			type = $field.attr('max') ? 'max' : 'min',
+			typeValue = $field.attr(type);
 
-		form.find('[maxlength]').each(function(element){
-			var length = parseInt(element.getAttribute('maxlength'));
+		if (fieldValue && typeValue !== null) {
+			var numericTypeValue = parseInt(typeValue, 10),
+				numericFieldValue = parseInt(fieldValue, 10);
 
-			if ($(element).val() && $(element).val().length > length) {
-				var parent = element.parentElement;
-				appendMessageWithArgs($(parent), validationMessages.maxlength, length);
-				valid = false;
+			if (isNaN(numericFieldValue) || condition(numericFieldValue, numericTypeValue)) {
+				return messageWithArgs(validationMessages[type], typeValue);
 			}
-		});
+		}
+	};
 
-		form.find('[pattern]').each(function(element){
-			var pattern = new RegExp('^' + element.getAttribute('pattern') + '$');
+	var getMinValidate = function($field) {
+		if (!$field.attr('min')) {
+			return false;
+		}
 
-			if ($(element).val() && !pattern.test($(element).val())) {
-				var parent = element.parentElement;
-				appendMessage($(parent), validationMessages.pattern);
-				valid = false;
-			}
+		return getMaxOrMinMessage($field, function(numericFieldValue, numericTypeValue) {
+			return numericFieldValue < numericTypeValue;
 		});
+	};
 
-		form.find('[type="email"]').each(function(element){
-			var pattern = new RegExp(/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/);
-
-			if (($(element).val()) && !pattern.test($(element).val())) {
-				var parent = element.parentElement;
-				appendMessage($(parent), validationMessages.email);
-				valid = false;
-			}
-		});
+	var getMaxValidate = function($field) {
+		if (!$field.attr('max')) {
+			return false;
+		}
 
-		form.find('[type="url"]').each(function(element){
-			if ($(element).val() && !element.validity.valid) {
-				var parent = element.parentElement;
-				appendMessage($(parent), validationMessages.url);
-				valid = false;
-			}
+		return getMaxOrMinMessage($field, function(numericFieldValue, numericTypeValue) {
+			return numericFieldValue > numericTypeValue;
 		});
+	};
 
-		if (!valid){
-			form.addClass('has-errors');
+	var getMaxLengthValidate = function($field) {
+		var length = $field.attr('maxlength'),
+			fieldValue = $field.val();
 
-			var visibleErrorFields = form.find('.error input, .error textarea').filter(function(field) {
-				return $(field).attr('type') !== 'hidden';
-			});
-
-			addEventToClearErrorMessages(visibleErrorFields);
-			this.focus(visibleErrorFields.first());
+		if (fieldValue && $field.attr('maxlength') !== null) {
+			if (fieldValue.length > length) {
```

Comment:
This couldn't be like these: `if (fieldValue && $field.attr('maxlength') !== null && fieldValue.length > length) {` ?

@tisvasconcelos